### PR TITLE
CI(eos_cli_config_gen): Restructured the molecule host vars for `aaa`, `local-users`, `radius-server` and `tacacs-server`

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Restructured the molecule host vars for `aaa`, `local-users`, `radius-server` and `tacacs-server`

## Related Issue(s)

Fixes #https://github.com/aristanetworks/avd-internal/issues/171

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Restructured the molecule host vars for `aaa`, `local-users`, `radius-server` and `tacacs-server`

## How to test
Run eos_cli_config_gen molecule then run tox command to generate the coverage report.

molecule converge -s eos_cli_config_gen
tox -e coverage,report

| Module | Coverage | Comment |
| ------- | ---------- | ---------- |
| eos/aaa-accounting.j2 | 95 | 8 partial |
| documentation/aaa-accounting.j2 | 98 | 3 partial |
| eos/aaa-authentication.j2 | 99 | 1 partial |
| documentation/aaa-authentication.j2 | 97 | 2 partial |
| eos/aaa-authorization.j2 | 100 | - |
| documentation/aaa-authorization.j2 | 99 | 1 partial |
| eos/aaa-root.j2 | 91 | deprecated key |
| documentation/aaa-server-groups.j2 | 100 | - |
| eos/aaa-server-groups.j2 | 96 | 2 partial |
| eos/local-users.j2 | 99 | 1 partial |
| documentation/local-users.j2 | 100 | - |
| eos/radius-server.j2 | 98 | 2 partial |
| documentation/radius-server.j2 | 99 | 1 partial |
| eos/tacacs-servers.j2 | 99 | 1 partial |
| documentation/tacacs-servers.j2 | 100 | - |
### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/radius-server.yml [hunk 1]
```diff
@@ -1,11 +1,20 @@
----
+### Radius Servers
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/radius-server.yml]:
```suggestion
---
### Radius Servers
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/tacacs-servers.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+### Tacacs+ Servers
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/tacacs-servers.yml]:
```suggestion
---
### Tacacs+ Servers
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-accounting-2.yml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+## AAA Accounting for none console
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-accounting-2.yml]:
```suggestion
---
## AAA Accounting for none console
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/local-users.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+### Local Users ###
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/local-users.yml]:
```suggestion
---
### Local Users ###
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+### AAA Server Groups
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml]:
```suggestion
---
### AAA Server Groups
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-root.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+### AAA Root ###
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-root.yml]:
```suggestion
---
### AAA Root ###
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-authorization.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+## AAA Authorization
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-authorization.yml:2]:
```suggestion
---
## AAA Authorization
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-accounting.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
+## AAA Accounting
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-accounting.yml:2]:
```suggestion
---
## AAA Accounting
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/radius-server-2.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/radius-server-2.yml:1]: All the files with -2 should be renamed to just their base name, since they are already under host2.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-accounting.yml (hunk 8 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-accounting.yml:2]: updated all

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/radius-server-2.yml
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/radius-server-2.yml:1]: changed the names

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+   server 10.10.10.249
+!
+aaa group server tacacs+ TACACS2
+   server 192.168.10.157 vrf mgt
+   server 10.10.10.248
+!
+aaa authentication login default group TACACS local
+aaa authentication login console local
+aaa authentication enable default group TACACS local
+aaa authentication dot1x default group TACACS
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg]: @alexeygorbunov @gmuloc @ClausHolbechArista 
Need help to configure this command. It require to create the dot1x group first but am unable to get it in my lab.
s1-host1(config)#aaa authentication dot1x default group TACACS
% group TACACS does not support dot1x authentication

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+aaa authentication policy on-failure log
+aaa authentication policy on-success log
+aaa authentication policy local allow-nopassword-remote-login
+aaa authentication policy lockout failure 3 window 900 duration 300
+aaa authorization policy local default-role network-admin
+aaa authorization serial-console
+aaa authorization dynamic dot1x additional-groups group radius group RADIUS1
+aaa authorization exec default group TACACS local
+aaa authorization commands all default group TACACS
+aaa authorization commands 5 default group radius
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: Getting this while configure the command on lab
% Authorization denied for command 'aaa authorization commands 5 default group radius': connect: Network is unreachable
@alexeygorbunov @gmuloc @ClausHolbechArista

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: The line above enabled command authorization towards the TACACS server you have configured.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg (hunk 9 above)
Comment (alexeygorbunov) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg]: I think TACACS+ is simply not supported by dot1x:
```
cdv116...19:43:51(config-s-s1)#aaa authentication dot1x default group ?
  WORD    Server-group name
  radius  Use list of all defined RADIUS hosts
```

https://www.arista.com/en/um-eos/eos-control-plane-security

```
Configuring 802.1X Port Security
Basic steps to implementing 802.1X Port-based Network Access Control and RADIUS accounting on the switch:

A RADIUS server is required on one or more of your network servers or management stations. 802.1X is not supported with the TACACS+ authentication protocol.
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg (hunk 10 above)
Comment (alexeygorbunov) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: yeap, you would either need to specify fallback method (like `local`) or test all this inside config session

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg (hunk 9 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg]: Changed to Radius group now its working

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/tacacs-servers.yml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+---
+## Tacacs+ Servers
+tacacs_servers:
+  policy_unknown_mandatory_attribute_ignore: true
+  hosts:
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/tacacs-servers.yml:4]: `hosts` is not a required key, but its not checked for defined in j2 template so it throws error if not defined.
need to fix the j2 template (may be in separate PR)
(found it by test coverage, it shows `hosts` always true)

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml [hunk 12]
```diff
@@ -0,0 +1,45 @@
+---
+## AAA Server Groups
+aaa_server_groups:
+  - name: TACACS
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml:4]: The `type` key should be `required` in schema.
Also, the same issue as tacacs server -  `servers` key is not required but its not checked in template so it throws error if not defined.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    type: tacacs+
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml:5]: The `type` key should be `required` in schema.
Also, the same issue as tacacs server - `servers` key is not required but its not checked for defined in template so it throws error if not defined. (need to update j2 template)

Comment (alexeygorbunov) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml:5]: > The type key should be required in schema.

Agree as J2 templates unconditionally require it to be set and EOS explicitly requires a valid value as well
```
cdv116...19:43:51(config-s-s1)#aaa group server ?
  ldap     LDAP server-group definition
  radius   RADIUS server-group definition
  tacacs+  TACACS+ server-group definition
cdv116...19:43:51(config-s-s1)#aaa group server
```

> Also, the same issue as tacacs server - servers key is not required but its not checked in template so it throws error if not defined.

EOS technically allows to create AAA group without assigning any servers to it, so making `aaa_server_groups[].servers` a required key may not align with EoS behavior.
```EOS
cdv116...13:26:32(config-s-s1)#show session-config diffs
--- system:/running-config
+++ session:/s1-session-config
+aaa group server radius EMPTY-GROUP
cdv116...13:26:37(config-s-s1)#commit
cdv116...13:26:40#show run sec aaa group server radius
aaa group server radius EMPTY-GROUP
cdv116...13:26:52#
```

Therefore i guess we need to adjust documentation template https://github.com/aristanetworks/avd/blob/3cac535c16978abb099f159b0bd6839536a9108c/python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/aaa-server-groups.j2#L16 to make sure it checks for the presence of `aaa_server_group.servers`. And even if not set - we may still need to render group name and type.

At the same time - each element within `aaa_server_groups[].servers[]` must have a valid `server` attribute set, as AAA group member in EoS must have a valid IP or FQDN defined. Therefore `aaa_server_groups[].servers[].server` should probably be a required key.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host2.cfg [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+no aaa root
+!
+transceiver qsfp default-mode 4x10G
+!
+hostname host2
+!
+radius-server attribute 32 include-in-access-req format myformat
+!
+tacacs-server policy unknown-mandatory-attribute ignore
+tacacs-server host 10.10.10.157
```
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host2.cfg]: > tacacs-server host 10.10.10.157

Should we change IP of this TACACS and move this to `host1` (as it seem to not overlap with anything there as long as we keep IP unique)? 

> tacacs-server policy unknown-mandatory-attribute ignore

This command may be OK to be moved under `host1` as well

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+!
+aaa group server radius RADIUS2
+   server 10.10.10.157 vrf mgt
+   server 10.10.10.249
+!
+tacacs-server timeout 10
+tacacs-server host 10.10.10.157 single-connection vrf mgt key 7 071B245F5A
+tacacs-server host 10.10.10.249 timeout 23 key 7 071B245F5A
+tacacs-server host 10.10.10.158 key 7 071B245F5A
+tacacs-server host 10.10.10.159 key 8a $kUVyoj7FVQ//yw9D2lbqjA==$kxxohBiofI46IX3pw18KYQ==$DOOM0l9uU4TrQt2kyA7XCKtjUA==
```
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:49]: This command seems to be accepted OK on cEOS/vEOS without any `common key`-related modifications. At the same time - EoS by default uses `type7` "hashing" and processed config will return `type7` hash:

```
ag-dc1-leaf1a(config-s-s6)#tacacs-server host 10.10.10.159 key 8a $kUVyoj7FVQ//yw9D2lbqjA==$kxxohBiofI46IX3pw18KYQ==$DOOM0l9uU4TrQt2kyA7XCKtjUA==
...
ag-dc1-leaf1a(config-s-s6)#show session-config diffs 
--- system:/running-config
+++ session:/s6-session-confi
+tacacs-server host 10.10.10.159 key 7 094041071E1116110A0F17392E363E362729161E
```

Just to keep things consistent we may want to move this config line to `host2` and configure it together with the following lines (to make sure it's not converted from `8a` to `7`):

```EOS
management security
  password encryption reversible aes-256-gcm
tacacs-server host 10.10.10.159 key 8a $kUVyoj7FVQ//yw9D2lbqjA==$kxxohBiofI46IX3pw18KYQ==$DOOM0l9uU4TrQt2kyA7XCKtjUA==
```

```yaml
management_security:
  password:
    encryption_reversible: aes-256-gcm
tacacs_servers:
  policy_unknown_mandatory_attribute_ignore: true
  hosts:
    - host: 10.10.10.159
      key: "$kUVyoj7FVQ//yw9D2lbqjA==$kxxohBiofI46IX3pw18KYQ==$DOOM0l9uU4TrQt2kyA7XCKtjUA=="
      key_type: "8a"
```

`password encryption reversible aes-256-gcm` command will change "hashing" for all `type7`s - therefore should move this to `host2` to not impact all other passwords there

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml [hunk 16]
```diff
@@ -0,0 +1,3 @@
+---
+## AAA Root for default config
+generate_default_config: true
```
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml]: Do we need to generate default config here in regards to aaa/radius/tacacs?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/tacacs-servers.yml (hunk 11 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/tacacs-servers.yml:4]: Raised the issue 4649 to track it.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml (hunk 16 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml]: Yes in regards to aaa for improve the coverage.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml (hunk 13 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml:5]: Created an issue to fix this - #4650

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host2.cfg (hunk 14 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host2.cfg]: removed it from host2

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg (hunk 15 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:49]: moved the 8a password server to host2 plus added new file of management_security

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml (hunk 13 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host1/aaa-server-groups.yml:5]: raised issue 4650 to track it

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml (hunk 16 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml]: Please use the aaa_root knob instead of generate_default_config which is deprecated.

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/aaa-root.yml]: Thanks.
Removed the file from host2 as having converge issue for not defining the generate_default_config. As decided deprecated keys will handle separately.

**Review by alexeygorbunov — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg (hunk 10 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: @alexeygorbunov so is the plan to test this on config session in the lab? or will it cause issues when trying to apply the config?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: We have talked about testing with cloudvision which uses a session.

Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/host1.cfg:77]: Yes, the plan is to run these tests via CloudVision, so this will not be an issue.

**Review by carlbuchmann — APPROVED** [review]
LGTM - reviewed comments and compared coverage report from devel and coverage increased!
